desc/optimize/_constraint_wrappers.py

```diff
@@ -308,21 +373,15 @@ def __init__(
             "constraint should be instance of ObjectiveFunction." ""
         )
         for con in constraint.objectives:
-            if not isinstance(
-                con,
-                (
-                    ForceBalance,
-                    RadialForceBalance,
-                    HelicalForceBalance,
-                    CurrentDensity,
-                ),
-            ):
+            if not con._equilibrium:
```

Comment:
OK so this flag is true if it can't be used as a nonlinear constraint for the proximal projection? Does that only ever apply to the equilibrium objectives? If not, the flag should be renamed something more general/descriptive.

Reply:
The way things are currently written, the projection method only works for equilibrium constraints. In theory we could figure out "projections" for other constraints but that's not the goal here. The `Objective._equilibrium` flag is just meant to be a replacement for the few places where we would list out all the different objectives that define an equilibrium condition.